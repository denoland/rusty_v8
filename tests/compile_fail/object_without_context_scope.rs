// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  v8::scope!(scope, &mut isolate);
  let _object = v8::Object::new(&*scope);
}

fn mock<T>() -> T {
  unimplemented!()
}
