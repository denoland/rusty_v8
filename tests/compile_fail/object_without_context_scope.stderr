error[E0308]: mismatched types
 --> tests/compile_fail/object_without_context_scope.rs:6:33
  |
6 |   let _object = v8::Object::new(&*scope);
  |                 --------------- ^^^^^^^ expected `&PinnedRef<'_, HandleScope<'_>>`, found `&PinnedRef<'_, HandleScope<'_, ()>>`
  |                 |
  |                 arguments to this function are incorrect
  |
  = note: expected reference `&PinnedRef<'_, HandleScope<'_, v8::Context>>`
             found reference `&PinnedRef<'_, HandleScope<'_, ()>>`
note: associated function defined here
 --> src/object.rs
  |
  |   pub fn new<'s>(scope: &PinScope<'s, '_>) -> Local<'s, Object> {
  |          ^^^
